gufe/components/explicitmoleculecomponent.py

```diff
@@ -0,0 +1,124 @@
+    """
+    Determine the correct name from the rdkit.Chem.Mol and the user-provided
+    name; ensure that is set in the rdkit representation.
+    """
+    try:
+        rdkit_name = mol.GetProp("_Name")
+    except KeyError:
+        rdkit_name = ""
+
+    try:
+        rdkit_name = mol.GetProp("ofe-name")
+    except KeyError:
+        pass
+
+    if name and rdkit_name and rdkit_name != name:
+        warnings.warn(
+            f"SmallMoleculeComponent being renamed from {rdkit_name}"
+            f"to {name}."
+        )
+    elif name == "":
+        name = rdkit_name
+
+    mol.SetProp("ofe-name", name)
+    return name
+
+
+def _ensure_ofe_version(mol: RDKitMol):
+    """Ensure the rdkit representation has the current version associated"""
+    mol.SetProp("ofe-version", __version__)
+
+
+class ExplicitMoleculeComponent(Component):
+    """Base class for explicit molecules.
+
+    This provides basic serialization and conversion to different
+    representations. Specific file formats, such as SDF for small molecules
+    or PDB for proteins, should be implemented in subclasses.
+
+    We default to use SMILES as the basis of the hash, but this can be
+    overridden using the `_hashmol` method.
+    """
+
+    def __init__(self, rdkit: RDKitMol, name: str = ""):
+        name = _ensure_ofe_name(rdkit, name)
+        _ensure_ofe_version(rdkit)
+        conformers = list(rdkit.GetConformers())
+        if not conformers:
+            raise ValueError("Molecule was provided with no conformers.")
+
+        n_confs = len(conformers)
+        if n_confs > 1:
+            warnings.warn(
+                f"Molecule provided with {n_confs} conformers. "
+                f"Only the first will be used."
+            )
+
+        self._rdkit = rdkit
+        self._hash = self._hashmol(name=name)
+
+    def __hash__(self):
+        return hash(self._hash)
+
+    def __eq__(self, other):
+        return hash(self) == hash(other)
+
+    def _hashmol(self, name):
+        return hashmol(self._rdkit, name=name)
+
+    def _defaults(self):
+        return super()._defaults()
+
+    # Props
+    @property
+    def name(self) -> str:
+        return self._hash.name
+
+    @property
+    def smiles(self) -> str:
+        return self._hash.smiles
+
+    @property
+    def total_charge(self):
+        return Chem.GetFormalCharge(self._rdkit)
+
+    # TO
+    def to_rdkit(self) -> RDKitMol:
+        """Return an RDKit copied representation of this molecule"""
+        return Chem.Mol(self._rdkit)
+
+    def to_json(self):
+        return json.dumps(self.to_dict())
+
+    # From
+    @classmethod
+    def from_json(cls, json_str):
+        dct = json.loads(json_str)
+        return cls.from_dict(dct)
+
+    @classmethod
+    def from_rdkit(cls, rdkit: RDKitMol, name: str = ""):
+        """Create a SmallMoleculeComponent, copying from an RDKit Mol"""
+        return cls(rdkit=Chem.Mol(rdkit), name=name)
+
+    # Not Implemented - interface Functions:
```

Comment:
hmmm true, but this would look like this:
![image](https://user-images.githubusercontent.com/12428005/191008676-39ea9428-7357-417a-bc4e-3b0f63d3046b.png)

And if I understand the code correctly the behaviour would change right :)

Reply:
The decorator [`@abc.abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod) means that you need to provide an implementation in a subclass -- it will raise an error if you try to create an instance of a subclass that doesn't instantiate this.

That said, it is technically possible to call this as `GufeTokenizable._to_dict(some_obj)`, which would currently return None instead of raising a `NotImplementedError`. I'd say that is incorrect behavior in `GufeTokenizable`, and should be changed there. That one is an edge case where I can't imagine it mattering in practice. However, with `GufeTokenizable._from_dict`, you can actually call `GufeTokenizable._from_dict({})` without error, and you would normally call a classmethod without needing an instance, so this is worth fixing. (Not as part of this PR: I'll fix it separately.) 

However, back to the original point for this PR: you don't need to add these functions here, and it's actually problematic since you don't have them decorated with `abc.abstractmethod`.)